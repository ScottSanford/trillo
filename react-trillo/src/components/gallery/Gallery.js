import React from 'react'
import './Gallery.css'

export default function Gallery({ images }) {

  const photos = images.map(image => <Photo photo={image} />)

  return (
    <div className="gallery">
      {photos}
    </div>
  )
}

function Photo({ photo }) {

  const { img, description, caption } = photo

  return (
    <figure className="gallery__item">
      <img src={img} alt={description} className="gallery__photo" />
      <figcaption className="gallery__caption">{caption}</figcaption>
    </figure>
  )
}