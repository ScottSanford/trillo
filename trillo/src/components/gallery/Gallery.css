/* GALLERY */
.gallery {
    display: flex;
}

.gallery__item {
    position: relative;
}

.gallery__photo {
    display: block;
    transition: all 0.3s;
    width: 100%; /* always define a % width for responsive images */
}

.gallery__photo:hover + .gallery__caption {
    opacity: 1;
    pointer-events: none; /* removes flickering */
    top: 50%;
    transform: translate(-50%, -50%);
    visibility: visible;
}

.gallery__caption {
    color: var(--color-white);
    font-weight: 600;
    font-size: 2.5rem;
    left: 50%;
    opacity: 0;
    position: absolute;
    text-align: center;
    top: 0%;
    transform: translate(-50%, -50%);
    transition: all 0.2s;
    visibility: hidden;
    width: 100%;
}