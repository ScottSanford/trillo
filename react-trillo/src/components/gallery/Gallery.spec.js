import React from 'react'
import { shallow } from 'enzyme'
import Gallery from './Gallery'

describe('Gallery Component', () => {

	const minProps = {
		images: []
	}

	it('should match the snapshot', () => {
		const wrapper = shallow(<Gallery {...minProps} />)
		expect(wrapper).toMatchSnapshot()
	})

	it('should have 3 images in the gallery', () => {
		const images = [
			{ img: '1.jpg', description: '1', caption: '1' },
			{ img: '2.jpg', description: '2', caption: '2' },
			{ img: '3.jpg', description: '3', caption: '3' },
		]
		const wrapper = shallow(<Gallery {...minProps} images={images} />)
		expect(wrapper.find('.gallery').children()).toHaveLength(3)
	})

	it('should return null if no images are passed', () => {
		const images = []
		const wrapper = shallow(<Gallery {...minProps} images={images} />)
		expect(wrapper.find('.gallery').children()).toHaveLength(0)
	})
})