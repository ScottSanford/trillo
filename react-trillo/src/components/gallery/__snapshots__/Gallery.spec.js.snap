// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery Component should match the snapshot 1`] = `
<div
  className="gallery"
/>
`;
